Server/JokesOnYou.Web.Api/Controllers/JokesController.cs

```diff
@@ -29,5 +31,26 @@ public async Task<ActionResult<IEnumerable<JokeReplyDto>>> GetAllJokesAsync()
             return Ok(jokeDtos);
         }
 
+        [AllowAnonymous]
+        [HttpGet("{id}")]
+        public async Task<JokeReplyDto> GetJokeById(int id)
+        {
+            var jokeReplyDto = await _jokesService.GetJokeReplyById(id);
+            return jokeReplyDto;
+        }
+
+        // Create the whole line for Update Joke API call.
+        // UpdateJoke inside controller HTTP put
+        // UpdateJoke inside JokeService
+        // UpdateJoke inside JokeRepository
+        // Use AutoMapper if you can.
+
+        [HttpPatch]
+        [AllowAnonymous]
+        public async Task<ActionResult<JokeUpdateDTO>> UpdateJoke(JokeUpdateDTO jokeUpdateDTO)
+        {
+            await _jokesService.UpdateJoke(jokeUpdateDTO);
+            return Ok(); 
```

Comment:
Use Nocontent() instead, it means the request was successfull but we have nothing to send back